ArchiSteamFarm.OfficialPlugins.Monitoring/MonitoringPlugin.cs

```diff
@@ -0,0 +1,239 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+using ArchiSteamFarm.Core;
+using ArchiSteamFarm.IPC.Integration;
+using ArchiSteamFarm.OfficialPlugins.Monitoring.Localization;
+using ArchiSteamFarm.Plugins;
+using ArchiSteamFarm.Plugins.Interfaces;
+using ArchiSteamFarm.Steam;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry.Metrics;
+using SteamKit2;
+
+namespace ArchiSteamFarm.OfficialPlugins.Monitoring;
+
+[Export(typeof(IPlugin))]
+[SuppressMessage("ReSharper", "MemberCanBeFileLocal")]
+internal sealed class MonitoringPlugin : OfficialPlugin, IWebServiceProvider, IGitHubPluginUpdates, IASF, IDisposable {
+	[JsonInclude]
+	[Required]
+	public override string Name => nameof(MonitoringPlugin);
+
+	[JsonInclude]
+	[Required]
+	public override Version Version => typeof(MonitoringPlugin).Assembly.GetName().Version ?? throw new InvalidOperationException(nameof(Version));
+
+	private bool Enabled => (ASF.GlobalConfig?.IPC ?? false) && (Config?.Enabled ?? MonitoringConfig.DefaultEnabled);
+
+	private const string MeterName = nameof(ArchiSteamFarm);
+
+	private const string MetricNamePrefix = "asf";
+
+	private Meter? Meter;
+
+	public void OnConfiguringServices(IServiceCollection services) {
+		ArgumentNullException.ThrowIfNull(services);
+
+		if (!Enabled) {
+			return;
+		}
+
+		services.AddOpenTelemetry().WithMetrics(
+			static builder => {
+				builder.AddPrometheusExporter(static config => config.ScrapeEndpointPath = "/Api/metrics");
+				builder.AddMeter("Microsoft.AspNetCore.Hosting");
+				builder.AddMeter("Microsoft.AspNetCore.Server.Kestrel");
+				builder.AddMeter(MeterName);
+
+				builder.AddView(
+					"http.server.request.duration",
+					new ExplicitBucketHistogramConfiguration {
+						Boundaries = [0, 0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1, 2.5, 5, 7.5, 10]
+					}
+				);
+			}
+		);
+	}
+
+	public void OnConfiguringApplication(IApplicationBuilder app) {
+		ArgumentNullException.ThrowIfNull(app);
+
+		if (!Enabled) {
+			return;
+		}
+
+		app.UseEndpoints(static builder => builder.MapPrometheusScrapingEndpoint());
+	}
+
+	[JsonInclude]
+	private MonitoringConfig? Config { get; set; }
+
+	public Task OnASFInit(IReadOnlyDictionary<string, JsonElement>? additionalConfigProperties = null) {
+		if (additionalConfigProperties == null) {
+			return Task.CompletedTask;
+		}
+
+		MonitoringConfig? config = null;
+
+		foreach ((string configProperty, JsonElement configValue) in additionalConfigProperties) {
+			try {
+				config = configProperty switch {
+					nameof(Monitoring) => configValue.Deserialize<MonitoringConfig>(),
+					_ => config
+				};
+			} catch (Exception e) {
+				ASF.ArchiLogger.LogGenericException(e);
+				ASF.ArchiLogger.LogGenericWarning(string.Format(CultureInfo.CurrentCulture, Strings.PluginDisabledInConfig, nameof(MonitoringPlugin)));
+
+				return Task.CompletedTask;
+			}
+		}
```

Comment:
`additionalConfigProperties.TryGetValue(…)` would be much less verbose and more performant

Reply:
You are right. We should probably also fix that in `ExamplePlugin`